app/Actions/Album/Create.php

```diff
@@ -68,6 +62,32 @@ private function set_parent(Album $album, ?Album $parentAlbum): void
 		}
 	}
 
+	/**
+	 * Set up the permissions.
+	 *
+	 * @param Album      $album
+	 * @param Album|null $parentAlbum
+	 *
+	 * @return void
+	 *
+	 * @throws UnexpectedException
+	 * @throws ConfigurationKeyMissingException
+	 */
+	private function set_permissions(Album $album, ?Album $parentAlbum): void
+	{
+		$defaultProtectionType = Configs::getValueAsEnum('default_album_protection', DefaultAlbumProtectionType::class);
+
+		if ($defaultProtectionType === DefaultAlbumProtectionType::PUBLIC) {
+			$album->access_permissions()->saveMany([AccessPermission::ofPublic()]);
+		}
+
+		if ($defaultProtectionType === DefaultAlbumProtectionType::INHERIT && $parentAlbum !== null) {
+			$album->access_permissions()->saveMany($this->copyPermission($parentAlbum));
+		}
+
+		$this->fixPermissionIfNotOwner($album);
```

Comment:
Does the fix verb is adequate ? Maybe we could have something like "grantFullPermissionsToNewOwner"